sunbird-es-utils/src/main/java/org/sunbird/common/ElasticSearchTcpImpl.java

```diff
@@ -0,0 +1,652 @@
+   */
+  @Override
+  public Future<String> save(String index, String identifier, Map<String, Object> data) {
+    long startTime = System.currentTimeMillis();
+    Promise<String> promise = Futures.promise();
+    ProjectLogger.log(
+        "ElasticSearchTcpImpl:save: method started at ==" + startTime + " for Index " + index,
+        LoggerEnum.PERF_LOG);
+    if (StringUtils.isBlank(identifier) || StringUtils.isBlank(index)) {
+      ProjectLogger.log(
+          "ElasticSearchTcpImpl:save: Identifier value is null or empty ,not able to save data.",
+          LoggerEnum.ERROR);
+      promise.success("ERROR");
+      return promise.future();
+    }
+    try {
+      data.put("identifier", identifier);
+      IndexResponse response =
+          ConnectionManager.getClient().prepareIndex(index, _DOC, identifier).setSource(data).get();
+      ProjectLogger.log(
+          "ElasticSearchTcpImpl:save: "
+              + "Save value=="
+              + response.getId()
+              + " "
+              + response.status(),
+          LoggerEnum.INFO.name());
+      ProjectLogger.log(
+          "ElasticSearchTcpImpl:save: method end at =="
+              + System.currentTimeMillis()
+              + " for INdex "
+              + index
+              + " ,Total time elapsed = "
+              + ElasticSearchHelper.calculateEndTime(startTime),
+          LoggerEnum.PERF_LOG);
+      promise.success(response.getId());
+      return promise.future();
+    } catch (Exception e) {
+      ProjectLogger.log(
+          "ElasticSearchTcpImpl:save: Error while saving index " + index + " id : " + identifier,
+          e);
+      ProjectLogger.log(
+          "ElasticSearchTcpImpl:save: method end at =="
+              + System.currentTimeMillis()
+              + " for Index "
+              + index
+              + " ,Total time elapsed = "
+              + ElasticSearchHelper.calculateEndTime(startTime),
+          LoggerEnum.PERF_LOG);
+      promise.failure(e);
+      promise.success("");
+    }
+    return promise.future();
+  }
+
+  /**
+   * This method will provide data form ES based on incoming identifier. we can get data by passing
+   * index and identifier values , or all the three
+   *
+   * @param identifier String
+   * @return Map<String,Object> or null
+   */
+  @Override
+  public Future<Map<String, Object>> getDataByIdentifier(String index, String identifier) {
+    long startTime = System.currentTimeMillis();
+    Promise<Map<String, Object>> promise = Futures.promise();
+    ProjectLogger.log(
+        "ElasticSearchTcpImpl:getDataByIdentifier: method started at =="
+            + startTime
+            + " for index "
+            + index,
+        LoggerEnum.PERF_LOG);
+    GetResponse response = null;
+    if (StringUtils.isBlank(index) || StringUtils.isBlank(identifier)) {
+      ProjectLogger.log(
+          "ElasticSearchTcpImpl:getDataByIdentifier: Invalid request is coming.", LoggerEnum.INFO);
```

Comment:
LoggerEnum.INFO to LoggerEnum.INFO.name()
info message should tell waht value was null or empty

Reply:
done